client/task-list/index.js

```diff
@@ -90,34 +82,29 @@ export class TaskDashboard extends Component {
 		} = this.props;
 		const { isCartModalOpen } = this.state;
 		const allTasks = this.getAllTasks();
-		const { extension: extensionTasks, setup: setupTasks } = this.groupBy(
-			allTasks,
-			'type'
-		);
+		const { extension: extensionTasks, setup: setupTasks } = allTasks;
 
 		return (
 			<>
 				{ setupTasks && ! isSetupTaskListHidden && (
 					<TaskList
-						allTasks={ allTasks }
 						dismissedTasks={ dismissedTasks }
 						isTaskListComplete={ isTaskListComplete }
 						isExtended={ false }
 						query={ query }
-						specificTasks={ setupTasks }
+						tasks={ allTasks }
 						trackedCompletedTasks={ trackedCompletedTasks }
 					/>
 				) }
 				{ extensionTasks && ! isExtendedTaskListHidden && (
 					<TaskList
-						allTasks={ allTasks }
 						dismissedTasks={ dismissedTasks }
 						isExtendedTaskListComplete={
 							isExtendedTaskListComplete
 						}
 						isExtended={ true }
```

Comment:
I know these weren't introduced in this PR, but why does the `<TaskList/>` need to know about the "extended" task list concept? Can't it just receive a `isComplete` and `tasks` props?